lib/elixir_analyzer/test_suite/high_score.ex

```diff
@@ -0,0 +1,45 @@
+defmodule ElixirAnalyzer.TestSuite.HighScore do
+  @moduledoc """
+  This is an exercise analyzer extension module for the concept exercise Bird Count
+  """
+
+  use ElixirAnalyzer.ExerciseTest
+
+  feature "uses a module attribute to define the initial score" do
+    find :any
+    type :essential
+    comment ElixirAnalyzer.Constants.high_score_use_module_attribute()
+
+    form do
+      @_shallow_ignore 0
+    end
+  end
+
+  feature "uses the module attribute in add_player function head" do
+    find :any
+    type :essential
+    suppress_if "uses the module attribute in add_player function body", :pass
+    comment ElixirAnalyzer.Constants.high_score_use_module_attribute()
+
+    form do
+      def add_player(_ignore, _ignore, _ignore \\ @_ignore) do
+        _ignore
+      end
+    end
+  end
+
+  assert_call "uses the module attribute in add_player function body" do
+    type :essential
+    suppress_if "uses the module attribute in add_player function head", :pass
+    comment ElixirAnalyzer.Constants.high_score_use_module_attribute()
+    calling_fn module: HighScore, name: :add_player
+    called_fn name: :@
+  end
```

Comment:
@neenjaw I am stuck with this part. I want to allow the student to either use the module attribute in the function's head as a default argument, or in the function's body in any way that they want. I don't want to have false positives if they use the attribute in the body in a roundabout way (e.g. x = @initial_score; Map.put(scores, name, x)`, that's why I'm trying to use `assert_call` for this instead of `feature`.

This means that I need a cyclic dependency with `suppress_if`, and that doesn't work. Do you have any thoughts on if it's possible to modify the analyzer to allow cyclic dependencies with `suppress_if`?

Alternatively, I can give up on the idea of allowing usage of the module attribute in the function body and instead add a new check that will tell students to use a function head with default arguments.

Reply:
Hmm, haven't thought about that.  Let me ponder this case

Reply:
@neenjaw if you have no ideas, then I'm just going to disallow the usage of the module attribute in the function body by adding some analysis that checks that a default argument was used.

Reply:
Just haven't had time to work on this, but that's a good stopgap for now